### PR TITLE
Clean-up doxygen modules

### PR DESCRIPTION
**Pull request by afh**
Looking at the [Modules in Ledger's API documentation](http://ledger-cli.org/doc/api/modules.html) there appear to be some inconsistencies, e.g. there is a "General utilities" and a "Utils" module, `generate.h` declares `@addotgroup generate` although it later specifies `@ingroup resport`, etc.

This PR tries to clean this up.

@jwiegley do you agree with the proposed changes and could you possibly provide a brief description for the following modules:

- [ ] **Mathematical objects:** _TBA_
- [ ] **Extra Functionality:** _TBA_
- [ ] **Data representation:** _TBA_
- [ ] **Python API:** _TBA_
- [ ] **Reporting:** _TBA_
- [ ] **Value expressions:** _TBA_
- [ ] **General utilities:** _TBA_
- [ ] **Views:** _TBA_

### CODE REVIEW

**Comment by jwiegley** [comment]
I really don't have much to say about them, beyond the titles themselves.

**Comment by afh** [comment]
Thanks for the review, @jwiegley. Mind approving this PR if you agree with the changes, so it can be merged.

I'm happy to merge as is and possibly add a brief description to the doxygen modules as my understanding of each grows.

**Review by tbm — APPROVED** [review]
up to @afh

**Comment by afh** [comment]
Thanks for merging, @tbm, much appreciated 🙏

**Pull request merged**
